source/openqasm/parser/antlr/qasm_parser.py

```diff
@@ -0,0 +1,893 @@
+    def visitQuantumLoop(self, ctx: qasm3Parser.QuantumLoopContext):
+        if ctx.loopSignature().getChild(0).getText() == "while":
+            return WhileLoop(
+                while_condition=self.visit(ctx.loopSignature().expression()),
+                block=self.visit(ctx.quantumLoopBlock()),
+            )
+        else:  # For In Loop
+            return ForInLoop(
+                loop_variable=add_span(
+                    Identifier(ctx.loopSignature().Identifier().getText()),
+                    get_span(ctx.loopSignature().Identifier()),
+                ),
+                set_declaration=self.visit(ctx.loopSignature().setDeclaration()),
+                block=self.visit(ctx.quantumLoopBlock()),
+            )
+
+    def visitQuantumLoopBlock(self, ctx: qasm3Parser.QuantumLoopBlockContext):
+        if ctx.LBRACE():
+            return [self.visit(statement) for statement in ctx.quantumStatement()]
+        else:
+            return [self.visit(ctx.quantumStatement())]
+
+    @span
+    def visitQuantumDeclarationStatement(self, ctx: qasm3Parser.QuantumDeclarationStatementContext):
+        return self.visit(ctx.quantumDeclaration())
+
+    @span
+    def visitQuantumDeclaration(self, ctx: qasm3Parser.QuantumDeclarationContext):
+        return QubitDeclaration(
+            add_span(QubitDeclTypeName[ctx.getChild(0).getText()], get_span(ctx.getChild(0))),
+            add_span(
+                Qubit(
+                    ctx.Identifier().getText(),
+                ),
+                get_span(ctx.Identifier()),
+            ),
+            self.visit(ctx.designator()) if ctx.designator() else None,
+        )
+
+    @span
+    def visitDesignator(self, ctx: qasm3Parser.DesignatorContext):
+        return self.visit(ctx.expression())
+
+    @span
+    def visitStatement(self, ctx: qasm3Parser.StatementContext):
+        return self.visit(ctx.getChild(0))
+
+    @span
+    def visitAliasStatement(self, ctx: qasm3Parser.AliasStatementContext):
+        target = Identifier(name=ctx.Identifier().getText())
+        value = self.visit(ctx.indexIdentifier())
+        return AliasStatement(add_span(target, get_span(ctx.Identifier())), value)
+
+    @span
+    def visitExpressionStatement(self, ctx: qasm3Parser.ExpressionStatementContext):
+        return ExpressionStatement(self.visit(ctx.expression()))
+
+    @span
+    def visitQuantumStatement(self, ctx: qasm3Parser.QuantumStatementContext):
+        return self.visit(ctx.getChild(0))
+
+    @span
+    def visitQuantumInstruction(self, ctx: qasm3Parser.QuantumInstructionContext):
+        quantum_gate_call = ctx.quantumGateCall()
+        quantum_phase = ctx.quantumPhase()
+        quantum_measurement = ctx.quantumMeasurement()
+        quantum_reset = ctx.quantumReset()
+        quantum_barrier = ctx.quantumBarrier()
+        if quantum_gate_call:
+            return self.visit(quantum_gate_call)
+        elif quantum_phase:
+            return self.visit(quantum_measurement)
```

Comment:
This seems like a typo. `quantum_measurement` should be `quantum_phase` here.
```suggestion
            return self.visit(quantum_phase)
```